apps/examples/webserver_test/webserver_test_util.h

```diff
@@ -0,0 +1,248 @@
+#define HTTP_CONF_MAX_DIVIDED_PATH_LENGTH       32
+#define HTTP_CONF_MAX_SLASH_COUNT               32
+#define HTTP_CONF_MAX_QUERY_HANDLER_COUNT       64
+#define HTTP_CONF_MAX_ENTITY_LENGTH             2048
+
+#define HTTP_ERROR_400            "Bad Request"
+#define HTTP_ERROR_404            "Not Found"
+#define HTTP_ERROR_500            "Internal Server Error"
+
+#define WS_TEST_CONF_MAX_URL_SIZE       50
+#define WS_TEST_CONF_MAX_ENTITY_SIZE    2048
+#define WS_TEST_CONF_MAX_PHRASE_SIZE    50
+#define WS_TEST_CONF_MAX_MESSAGE_SIZE   2100
+
+#define WS_TEST_MODE_GET                     0
+#define WS_TEST_MODE_POST                    1
+#define WS_TEST_MODE_PUT                     2
+#define WS_TEST_MODE_DELETE                  3
+
+#define HTTPSTATUS_NONE            0
+#define HTTPSTATUS_OK              1
+#define HTTPSTATUS_MOVED           2
+#define HTTPSTATUS_ERROR           3
+
+#define ISO_nl                     0x0a
+#define ISO_cr                     0x0d
+#define ISO_space                  0x20
+
+#define WS_TEST_MAXHOSTNAME 128
+
+static const char HTTP11[] = "HTTP/1.1";
+static const char HTTPHOST[] = "host: ";
+static const char HTTPGET[] = "GET ";
+static const char HTTPPOST[] = "POST ";
+static const char HTTPPUT[] = "PUT ";
+static const char HTTPDELETE[] = "DELETE ";
+static const char HTTPCRNL[] = "\r\n";
+
+static const char HTTPFORM[] = "Content-Type: application/x-www-form-urlencoded";
+static const char HTTPCONTENTSIZE[] = "Content-Length: ";
+static const char HTTPCHUNKED[] = "Transfer-Encoding: chunked";
+static const char TRAILERHEADER[] = "Trailer: Expires";
+static const char EXPIRES[] = "Expires: Fri, 21 Oct 2022 07:28:00 GMT";
+
+static const char HEADERFIELD_CONNECT[] = "Connection";
+static const char HEADERFIELD_CLOSE[] = "close";
+static const char HEADERFIELD_KEEPALIVE[] = "keep-alive";
+static const char HEADERFIELD_USERAGENT[] = "User-Agent";
+static const char HEADERFIELD_TINYARA[] = "TinyARA";
+
+#define PRNT(...) \
+	do { \
+		printf("[WS_TEST]%s():%d: ", __func__, __LINE__); \
+		printf(__VA_ARGS__); \
+		printf("\n"); \
+	} while(0)
+
+#define FUNC_EN PRNT("entry")
+#define FUNC_EX PRNT("exit")
+
+enum {
+	HTTP_REQUEST_HEADER, HTTP_REQUEST_PARAMETERS, HTTP_REQUEST_BODY
+};
+
+typedef struct http_client_response_t *httprsp;
+typedef void (*wget_callback_t)(httprsp);
+
+#ifdef CONFIG_NET_SECURITY_TLS
+/**
+ * @brief HTTP client TLS structure.
+ */
+struct http_client_tls_t {
+	int client_fd;
+	mbedtls_ssl_context       tls_ssl;
+	mbedtls_net_context       tls_client_fd;
+
+	int                       tls_init;
+	mbedtls_ssl_config        tls_conf;
+	mbedtls_entropy_context   tls_entropy;
+	mbedtls_ctr_drbg_context  tls_ctr_drbg;
+	mbedtls_x509_crt          tls_rootca;
+	mbedtls_x509_crt          tls_clicert;
+	mbedtls_pk_context        tls_pkey;
+	mbedtls_ssl_session       tls_session;
+};
+#endif
+
+/**
+ * @brief SSL configure structure.
+ */
+struct http_client_ssl_config_t {
```

Comment:
ssl is a sort of tls. So it might require CONFIG_NET_SECURITY_TLS.
Furthermore you already add dependency to Kconfig. So those TLS configuration is not needed to source files.